### PR TITLE
Add an upgrade guide for v2

### PR DESCRIPTION
**Pull request by grossbart**
I started an upgrade guide for the v1 -> v2 transition, I hope it helps. Depending on how many questions are asked, this guide might evolve.

I hoped to do this earlier, but time is sparse at the moment, sigh …

### CODE REVIEW

**Comment by dakom** [comment]
Until it is merged - I guess it makes sense to add suggestions/questions here?

One that just bit me - seems there is no longer `O.getOrElseL()`, rather `O.getOrElse()` is _always_ lazy?

**Comment by grossbart** [comment]
I'll add a note to the document, the discussion happened here: https://github.com/gcanti/fp-ts/issues/823#issuecomment-500301156

tl;dr having a single API is preferrable, and always lazy is more general.

File: docs/introduction/upgrade-to-v2.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
++  "rulesDirectory": ["./node_modules/fp-ts/rules"],
+   "rules": {
++    "obsolete": true
+   }
+}
+```
+
+### Dependencies
+
+Make sure to always have a single version of `fp-ts` installed in your project. Multiple versions are known to cause `tsc` to hang during compilation. You can check the versions currently installed using `npm ls fp-ts`: make sure there's a single version and all the others are marked as `deduped`.
```
Comment (giogonzo) [inline docs/introduction/upgrade-to-v2.md]: I think this (tsc hanging) will be no longer the case with fp-ts >=2 ? I remember someone mentioning it in multiple occasions, but never verified myself. cc @gcanti

Comment (leemhenson) [reply docs/introduction/upgrade-to-v2.md]: If _all_ the versions you have installed at >= 2 then I _think_ the issue will no longer occur.

Comment (grossbart) [reply docs/introduction/upgrade-to-v2.md]: I think you're right, this was one of the motivations behind the removal of classes.

**Comment by dakom** [comment]
Suggest mentioning (as mentioned in https://github.com/gcanti/fp-ts/issues/903) that `pipe()` is helpful, even with only one function, for the following reasons:

1. type inference
2. allows for easier migration since args don't need to be flipped around

**Pull request merged**